packages/scene-composer/src/components/panels/scene-components/data-overlay/DataBindingMapEditor.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useCallback } from 'react';
+import { Box, Button, ExpandableSection, FormField, Input, SpaceBetween } from '@awsui/components-react';
+import { useIntl } from 'react-intl';
+import { isEmpty } from 'lodash';
+
+import { IValueDataBinding, IValueDataBindingProvider } from '../../../../interfaces';
+import { IDataOverlayComponentInternal } from '../../../../store';
+import { ValueDataBindingBuilder } from '../ValueDataBindingBuilder';
+import { generateUUID } from '../../../../utils/mathUtils';
+
+interface IDataBindingMapEditorProps {
+  valueDataBindingProvider: IValueDataBindingProvider | undefined;
+  component: IDataOverlayComponentInternal;
+  onUpdateCallback: (componentPartial: Partial<IDataOverlayComponentInternal>, replace?: boolean | undefined) => void;
+}
+
+export const DataBindingMapEditor: React.FC<IDataBindingMapEditorProps> = ({
```

Comment:
This component has too many responsibilities, and should be decomposed into smaller pieces. Use judgement, but pretty much every formfield/input combination can be it's own component responsible for it's own strings and behaviors, and it greatly simplifies testing and code structure.

Just apply single responsibility principle, each component should really only have one reason to change.

Reply:
Please break this one down into smaller pieces.